test/System/Angle.cpp

```diff
@@ -0,0 +1,204 @@
+            CHECK(sf::degrees(720.0f) == sf::Angle::Zero);
+            CHECK(sf::degrees(-360.0f) == sf::Angle::Zero);
+            CHECK(sf::degrees(-720.0f) == sf::Angle::Zero);
+        }
+
+        SUBCASE("operator!=")
+        {
+            CHECK(sf::Angle() != sf::radians(2.0f));
+            CHECK(sf::degrees(1.0f) != sf::radians(1.0f));
+            CHECK(sf::radians(0.0f) != sf::radians(0.1f));
+        }
+
+        SUBCASE("Unary operator-")
+        {
+            CHECK(-sf::Angle() == sf::Angle());
+            CHECK(-sf::radians(-1.0f) == sf::radians(1.0f));
+            CHECK((-sf::degrees(15.0f)).asDegrees() == Approx(345.0f));
+            CHECK((-sf::radians(1.0f)).asRadians() == Approx(5.2831853071f));
+        }
+
+        SUBCASE("operator+")
+        {
+            CHECK(sf::Angle() + sf::Angle() == sf::Angle());
+            CHECK(sf::Angle::Zero + sf::radians(0.5f) == sf::radians(0.5f));
+            CHECK((sf::radians(6.0f) + sf::radians(0.5f)).asRadians() == Approx(sf::radians(0.216815f).asRadians()));
+            CHECK((sf::radians(10.0f) + sf::radians(0.5f)).asRadians() == Approx(sf::radians(4.21681f).asRadians()));
+            CHECK((sf::degrees(900.0f) + sf::degrees(900.0f)).asRadians() == Approx(0.0f));
+        }
+
+        SUBCASE("operator+=")
+        {
+            sf::Angle angle;
+            angle += sf::degrees(15.0f);
+            CHECK(angle.asDegrees() == Approx(15.0f));
+            angle += sf::radians(10.0f);
+            CHECK(angle.asDegrees() == Approx(227.957795f));
+        }
+
+        SUBCASE("operator-")
+        {
+            CHECK(sf::Angle() - sf::Angle() == sf::Angle());
+            CHECK(sf::radians(1.0f) - sf::radians(0.5f) == sf::radians(0.5f));
+            CHECK(sf::Angle::Zero - sf::radians(0.5f) == sf::radians(-0.5f));
+            CHECK((sf::Angle::Zero - sf::radians(0.5f)).asRadians() == Approx(sf::radians(5.783185f).asRadians()));
+            CHECK((sf::degrees(900.0f) - sf::degrees(1.0f)).asRadians() == Approx(sf::degrees(179.0f).asRadians()));
+        }
+
+        SUBCASE("operator-=")
+        {
+            sf::Angle angle = sf::degrees(15.0f);
+            angle -= sf::degrees(15.0f);
+            CHECK(angle.asDegrees() == Approx(0.0f));
+            angle -= sf::radians(10.0f);
+            CHECK(angle.asRadians() == Approx(2.56637061f));
+        }
+
+        SUBCASE("operator*")
+        {
+            CHECK(sf::radians(0.0f) * 10.0f == sf::Angle::Zero);
+            CHECK((sf::degrees(10.0f) * 2.5f).asDegrees() == Approx(25.0f));
+            CHECK((sf::degrees(100.0f) * 10.0f).asDegrees() == Approx(280.0f));
+
+            CHECK(10.0f * sf::radians(0.0f) == sf::Angle::Zero);
+            CHECK((2.5f * sf::degrees(10.0f)).asDegrees() == Approx(25.0f));
+            CHECK((10.0f * sf::degrees(100.0f)).asDegrees() == Approx(280.0f));
+
+            CHECK(sf::radians(0.0f) * static_cast<sf::Int64>(10) == sf::Angle::Zero);
```

Comment:
This does not compile without the static cast. GCC complains:
```
error: ambiguous overload for ‘operator*’ (operand types are ‘sf::Angle’ and ‘int’)
note: candidate: ‘constexpr sf::Angle sf::operator*(sf::Angle, float)’
note: candidate: ‘constexpr sf::Angle sf::operator*(sf::Angle, sf::Int64)’
```

It looks like a problem.
Could operators just take `int` instead of `Int64`? Or maybe only accept `float` as for `Vector2f`.

Reply:
I think we should only have `float` overloads here.

Reply:
I removed those integer overloads since all they did was cast to float before doing the same calculations. Users can easily achieve the same thing on their own without us adding these ambiguous overloads.